compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -551,6 +551,10 @@ pub const BUILTIN_ATTRIBUTES: &[BuiltinAttribute] = &[
         may_dangle, Normal, template!(Word), WarnFollowing, dropck_eyepatch,
         "`may_dangle` has unstable semantics and may be removed in the future",
     ),
+    gated!(
+        rustc_auto_trait, Normal, template!(Word), WarnFollowing, auto_traits,
```

Comment:
This should already be gated with rustc_attrs, I don't see the point in having an extra gate. It should be moved to the other rustc attrs.

Reply:
+1 on removing `auto_traits` and making `#[rustc_auto_trait]` part of `rustc_attrs`. I didn't do that since people in the Zulip topic were proposing marking `auto_traits` internal instead of removing it.